### PR TITLE
Use new balance scanner contract

### PR DESCRIPTION
**Pull request by yabirgb**
This commit introduces a new balance scanner contracts. It is writting in vyper and is important because in case of a revert calling the balanceof for a contract it returns 0 instead of failing the transaction.

This change solves an issue where a bad token could make rotki not query balances properly.

Regarding the chunk size I've checked different sizes

```
chunk size, time, node, seed, comments

375, 5.452023983001709,  flashbots
375, 5.356801986694336,  flashbots , 42
375, 8.883646965026855,  cloudflare
375, 10.064039945602417, cloudflare
375, 6.45735502243042,   cloudflare
375, 5.277329921722412,  cloudflare, 80
375, 4.732897043228149,  ankr      , 90
500, 4.8045032024383545, ankr + flashbots, 90,  ankr and cloudlare had a single call out of gas
480, 5.05453896522522, ankr, 90, ankr had a single call out of gas. Cloudflare okey
460, 3.771683931350708, ankr + cloudflare, 90, only 1 request to ankr failed
460, 4.571718215942383, ankr + cloudflare, 90, only 1 request to ankr failed
460, 6.436861991882324, flashbots, 42
460, 3.726022243499756, flahbots + ankr, 80
460, 7.433700084686279, cloudflare + ankr, 460
```

and the optimal was 460. I'm under the impression that a token consumes a little bit more of gas and makes the transaction fail.

The checks were made with this test

```python
import pytest
from rotkehlchen.chain.ethereum.manager import EthereumManager
from rotkehlchen.types import ChecksumEvmAddress
from rotkehlchen.tests.utils.ethereum import wait_until_all_nodes_connected
from rotkehlchen.types import SupportedBlockchain
import time
import random

@pytest.mark.parametrize('ethereum_accounts', [[
    '0x3Ba6eB0e4327B96aDe6D4f3b578724208a590CEF',
]])
@pytest.mark.parametrize('ethereum_manager_connect_at_start', ['DEFAULT'])
def test_token_detection(
    ethereum_manager: EthereumManager,
    ethereum_accounts: list[ChecksumEvmAddress],
):
    random.seed(120)
    # nodes = ethereum_manager.node_inquirer.database.get_rpc_nodes(blockchain=SupportedBlockchain.ETHEREUM)
    # ethereum_manager.node_inquirer.connect_to_multiple_nodes(nodes)
    # wait_until_all_nodes_connected(
    #     connect_at_start=nodes,
    #     evm_inquirer=ethereum_manager.node_inquirer,
    # )
    start = time.time()
    tokens = ethereum_manager.tokens._query_new_tokens(
        addresses=ethereum_accounts,
    )
    # print(tokens)
    print(time.time() - start)
```

### CODE REVIEW

File: rotkehlchen/chain/evm/base_contracts.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import json
+from typing import Final
+
+from rotkehlchen.chain.evm.contracts import EvmContract
+from rotkehlchen.chain.evm.types import string_to_evm_address
+
+BALANCE_SCANNER: Final = EvmContract(  # defined here only until we merge bugfixes to develop  # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/base_contracts.py:7]: can you make a note to do this and not forget?

File: rotkehlchen/chain/evm/tokens.py [hunk 2]
```diff
@@ -64,7 +64,7 @@
 # tokens the benchmark will probably have to run again.
 
 
-OTHER_MAX_TOKEN_CHUNK_LENGTH = 590
+OTHER_MAX_TOKEN_CHUNK_LENGTH = 460
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/tokens.py:67]: Can you update the benchmark above and explain why this value is chosen? As the benchmark and the entire comment is for a completely different contract. It now needs to change.

File: rotkehlchen/tests/api/blockchain/test_optimism.py [hunk 3]
```diff
@@ -66,8 +66,7 @@ def test_add_optimism_blockchain_account(rotkehlchen_api_server: 'APIServer') ->
     now = ts_now()
     # now check that detecting tokens works
     optimism_tokens = (
-        Asset('eip155:10/erc20:0x4F604735c1cF31399C6E711D5962b2B3E0225AD3'),
-        Asset('eip155:10/erc20:0x94b008aA00579c1307B0EF2c499aD98a8ce58e58'),
+        Asset('eip155:10/erc20:0x0b2C639c533813f4Aa9D7837CAf62653d097Ff85'),
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/blockchain/test_optimism.py:69]: hmmm why did one token disappear from here?

File: rotkehlchen/tests/api/test_aave.py [hunk 4]
```diff
@@ -102,43 +102,22 @@ def mock_token_balances(addresses: 'Sequence[ChecksumEvmAddress]') -> TokenBalan
             'lending': {
                 A_DAI.identifier: {
                     'balance': Balance(
-                        amount=FVal('25319.644933920807083535'),
-                        usd_value=FVal('37979.4674008812106253025'),
+                        amount=FVal('21528.498187325929022298'),
```
Comment (LefterisJP) [inline rotkehlchen/tests/api/test_aave.py:105]: This (and other files with similar changes) changed due to re-recording VCR?

Comment (yabirgb) [reply rotkehlchen/tests/api/test_aave.py:105]: yes, values got updated

File: rotkehlchen/chain/evm/base_contracts.py (hunk 1 above)
Comment (yabirgb) [reply rotkehlchen/chain/evm/base_contracts.py:7]: Done here https://github.com/orgs/rotki/projects/11?pane=issue&itemId=88901029

File: docs/changelog.rst [hunk 5]
```diff
@@ -8,7 +8,8 @@ Changelog
 * :bug:`-` Fix the issue where some asset values show zero in the edit snapshot form.
 * :bug:`-` Fix the issue where the pagination for the account table resets to the first page when the user expands the account.
 * :bug:`8892` rotki will now correctly fetch Starknet token prices before May 2024 from Cryptocompare, when the ticker changed from STARK to STRK.
-* :bug:`-` The airdrops directory should no longer appear under the user directory in certain cirsumstances
+* :bug:`-` Fix an issue that caused the token detection to fail under some circumstances.
```
Comment (LefterisJP) [inline docs/changelog.rst:11]: ehmm ... dude ?

**Review by LefterisJP — APPROVED** [review]
LGTM

**Pull request merged**